wire-ios/Wire-iOS/Sources/Authentication/Interface/Helpers/TextFieldValidator.swift

```diff
@@ -35,10 +40,26 @@ final class TextFieldValidator {
 
     private func validatePasscode(text: String,
                                   kind: ValidatedTextField.Kind,
-                                  isNew: Bool) -> TextFieldValidator.ValidationError? {
+                                  isNew: Bool,
+                                  ruleSet: PasswordValidationRuleSet = .defaultRuleSet) -> TextFieldValidator.ValidationError? {
+
+        let currentRuleSet: PasswordRuleSet?
+        switch ruleSet {
+        case .guestLinkRuleSet:
+            currentRuleSet = PasswordRuleSet.guestLinkWithPasswordRuleSet
+        case .defaultRuleSet:
+            currentRuleSet = PasswordRuleSet.shared
+            // add other cases as required
```

Comment:
By adding these, we're making `TextFieldValidator` specific to guest links. Instead, if we inject the rule set, the text field validator doesn't need to know anything about what kind of rule set it is... it just uses it.

Reply:
So in this case you mean to not use an enum and work with the PasswordRuleSet directly?

Reply:
Yep. What benefit does this enum actually give us?

Reply:
As @KaterinaWire suggested to be more expandable in the future:

https://github.com/wireapp/wire-ios/pull/606#discussion_r1362076407

Reply:
Done!